### PR TITLE
rofiles-fuse: Support write/read_buf()

### PR DESCRIPTION
**Pull request by alexlarsson**
These allow us to avoid copying a lot of data around
in userspace. Instead we splice the data directly from
the fd to the destination fd.

### CODE REVIEW

File: src/rofiles-fuse/main.c [hunk 1]
```diff
@@ -345,6 +345,26 @@ callback_create(const char *path, mode_t mode, struct fuse_file_info *finfo)
 }
 
 static int
+callback_read_buf (const char *path, struct fuse_bufvec **bufp,
+                   size_t size, off_t offset, struct fuse_file_info *finfo)
+{
+  struct fuse_bufvec *src;
+
+  src = malloc (sizeof (struct fuse_bufvec));
```
Comment (cgwalters) [inline src/rofiles-fuse/main.c]: Did fuse really add an interface for this that forces us to malloc for every read?   Reading the source, I guess so, but man, that's ugly.

**Comment by cgwalters** [comment]
Reading the FUSE source, it looks like this interface was [added in 2010](https://github.com/libfuse/libfuse/commit/63322038) so I think we're safe to use this unconditionally.

@rh-atomic-bot r+  	009541e

**Comment by cgwalters** [comment]
It looks like (unsurprisingly) the lowlevel API is more efficient.  So we should look at porting to that at some point.

**Comment by alexlarsson** [comment]
That one is hard to back of a path-based backend though, it assumes a filesystem-like thing where you operate on individual inodes.

**Comment by alexlarsson** [comment]
However, if you want performance, you should use -o kernel_cache,entry_timeout=60,attr_timeout=60 or something as arguments to rofiles-fuse.

**Pull request closed without merging**